pyproject.toml

```diff
@@ -54,6 +54,7 @@ doc = [
     "sphinx-gallery",
     "sphinxext-opengraph",
     "statsmodels",
+    "pyarrow",
```

Comment:
`project.optional-dependencies` represent the `pip` as in `requirements.txt`. Basically, you would need to add it also in the `tool.pixi.feature.doc.dependencies` to have it for `pixi` as well.

But at the end, the documentation `doc` environment uses default + doc + optional so it should include `pyarrow` already.